code/parse/sexp.cpp

```diff
@@ -13980,12 +13984,17 @@ void sexp_tech_add_intel(int node, bool xstr)
 
 		// if we have an xstr, we already translated this node in the preloader, so just look it up
 		i = intel_info_lookup(name);
-		if (i >= 0)
-			Intel_info[i].flags |= IIF_IN_TECH_DATABASE;
+		if (i >= 0) {
+			if (add) {
+				Intel_info[i].flags |= IIF_IN_TECH_DATABASE;
+			} else {
+				Intel_info[i].flags &= IIF_IN_TECH_DATABASE;
+			}
+		}
 		else if (xstr)
-			Warning(LOCATION, "In tech-add-intel-xstr, entry XSTR(\"%s\", %d) invalid", name, id);
+			Warning(LOCATION, "In tech-%-intel-xstr, entry XSTR(\"%s\", %d) invalid", (add? "add" : "remove"), name, id);
```

Comment:
That should be `%s` to specify strings.  `%` by itself won't show anything.  Also, as a nitpick, put a space before the `?`.